src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/VariableFetchAnalyzer.php

```diff
@@ -531,26 +540,239 @@ public static function getGlobalType(string $var_id): Union
+                'PHP_AUTH_DIGEST'      => $non_empty_string_helper,
+                'PHP_AUTH_USER'        => $non_empty_string_helper,
+                'PHP_AUTH_PW'          => $non_empty_string_helper,
+                'AUTH_TYPE'            => $non_empty_string_helper,
+                'PATH_INFO'            => $non_empty_string_helper,
+                'ORIG_PATH_INFO'       => $non_empty_string_helper,
+                // misc from RFC not included above already http://www.faqs.org/rfcs/rfc3875.html
+                'CONTENT_LENGTH'       => $string_helper,
+                'CONTENT_TYPE'         => $string_helper,
+                // common, misc stuff
+                'FCGI_ROLE'            => $non_empty_string_helper,
+                'HOME'                 => $non_empty_string_helper,
+                'HTTP_CACHE_CONTROL'   => $non_empty_string_helper,
+                'HTTP_COOKIE'          => $non_empty_string_helper,
+                'HTTP_PRIORITY'        => $non_empty_string_helper,
+                'PATH'                 => $non_empty_string_helper,
+                'REDIRECT_STATUS'      => $non_empty_string_helper,
+                'REQUEST_SCHEME'       => $non_empty_string_helper,
+                'USER'                 => $non_empty_string_helper,
+                // common, misc headers
+                'HTTP_UPGRADE_INSECURE_REQUESTS' => $non_empty_string_helper,
+                'HTTP_X_FORWARDED_PROTO'         => $non_empty_string_helper,
+                'HTTP_CLIENT_IP'                 => $non_empty_string_helper,
+                'HTTP_X_REAL_IP'                 => $non_empty_string_helper,
+                'HTTP_X_FORWARDED_FOR'           => $non_empty_string_helper,
+                'HTTP_CF_CONNECTING_IP'          => $non_empty_string_helper,
+                'HTTP_CF_IPCOUNTRY'              => $non_empty_string_helper,
+                'HTTP_CF_VISITOR'                => $non_empty_string_helper,
+                'HTTP_CDN_LOOP'                  => $non_empty_string_helper,
+                // common, misc browser headers
+                'HTTP_DNT'                => $non_empty_string_helper,
+                'HTTP_SEC_FETCH_DEST'     => $non_empty_string_helper,
+                'HTTP_SEC_FETCH_USER'     => $non_empty_string_helper,
+                'HTTP_SEC_FETCH_MODE'     => $non_empty_string_helper,
+                'HTTP_SEC_FETCH_SITE'     => $non_empty_string_helper,
+                'HTTP_SEC_CH_UA_PLATFORM' => $non_empty_string_helper,
+                'HTTP_SEC_CH_UA_MOBILE'   => $non_empty_string_helper,
+                'HTTP_SEC_CH_UA'          => $non_empty_string_helper,
+            ]);
+
+            // generic case for all other elements
+            $detailed_type->previous_key_type = Type::getNonEmptyString();
+            $detailed_type->previous_value_type = Type::getString();
+
+            return new Union([$detailed_type]);
+        }
+
+        if ($var_id === '$_FILES') {
```

Comment:
Probably incorrect, since first array access is form input name, not "name" or "type" etc.

Reply:
https://www.php.net/manual/en/features.file-upload.post-method.php

Reply:
You didn't add a unit test for this check?